### ISSUE TITLE
Add aria2 or axel as optional downloader.

### ISSUE THREAD

**Issue by SlaterByte** [apt-mirror/apt-mirror#29]
# tl;dr

The backend downloader for apt-mirror is wget. Does it make sense to add support for [aria2](http://aria2.sourceforge.net/) or [axel](http://axel.alioth.debian.org/) as optional downloaders?
- axel is more light-weight and has less dependencies.
- aria2 is heavier with more features that may or may not be helpful in the case of apt-mirror.
# More

wget is a single-threaded downloader, though apt-mirror has a lot of code to spawn n-threads of wget. From what I can read in the code, it pulls down the indexes and then splits up the to-be-downloaded URIs into archive-urls.n which are fed to individual wget processes. Not bad, but **it only helps where each connection to the server is throttled**. It does nothing towards pulling the same file with multiple connections to one server, or to pull from multiple servers. The plus side of using wget is that it is installed by default on most Linux servers and is a well-known tool.

aria2 or axel, on the other hand, are multi-threaded downloaders to begin with. All the logic of multi-threading the download list can be handled by one of these instead of apt-mirror. Due to wget's ubiquity, you aren't going trash that code, but I imagine that adding support for aria2 or axel should be fairly straight-forward.

The main benefit is that these downloaders can pull the same file from multiple servers or the same file with multiple connections. apt-mirror with wget cannot do either. This would be a substantial benefit where we use apt-mirror with a really low-end connection.

Glad to see apt-mirror getting some polish!

**Comment by odyssey4me** [comment]
👍 This would be marvelous. This is something interesting I haven't tried yet, but could perhaps be integrated somehow: https://github.com/tatsuhiro-t/apt-metalink

**Comment by chutzimir** [comment]
I have been using aria2c for a while, and am much happier with it. Pushed #184, do give it a try.
